### PR TITLE
Upgrade Rust version

### LINKED ISSUE #536: Upgrade Rust to 1.47

### PR DESCRIPTION
**Pull request by junha1**
fixes #536

### CODE REVIEW

File: key/benches/ed25519.rs [hunk 1]
```diff
@@ -14,39 +14,31 @@
...
-        let message = Message::random();
-        let signature = sign(message.as_ref(), key_pair.private());
-        assert!(verify(&signature, message.as_ref(), key_pair.public()));
-    })
+fn ec25519_sign(c: &mut Criterion) {
+    c.bench_function("ec25519_sign", |b| {
+        b.iter(|| {
+            let key_pair: Ed25519KeyPair = Random.generate().unwrap();
+            let message = Message::random();
+            let _signature = sign(message.as_ref(), key_pair.private());
```
Comment (majecty) [inline key/benches/ed25519.rs:27]: Why did you remove the line below?
`         assert!(verify(&signature, message.as_ref(), key_pair.public())); `

**Review by majecty — CHANGES_REQUESTED** [review]

File: key/benches/ed25519.rs (hunk 1 above)
Comment (junha1) [reply key/benches/ed25519.rs:27]: It is from `ec25519_sign`, not `ed25519_sign_and_verify`. There was no assert in original code.
I split the commit to make it clear

**Review by majecty — APPROVED** [review]
